### PR TITLE
MP-9063 Refactor DtoDbContextGenerator to use Scriban template

### PR DESCRIPTION
**Pull request by msadikkoseaupa**

### CODE REVIEW

**Comment by gjorgji-grgovski-authoritypartners** [comment]
@msadikkoseaupa can you rebuild the whole solution, so that it re-generates the files and commits those too?
Also in the title of the PR, you need to have the MP ticket number.

File: src/Nox.Generator/Infrastructure/Persistence/DtoDbContextGenerator.cs [hunk 1]
```diff
@@ -22,115 +22,15 @@ public void Generate(SourceProductionContext context, NoxSolutionCodeGeneratorSt
...
-        code.EndBlock();
-        code.AppendLine();
+        const string className = $"DtoDbContext";
+        const string templateName = @"Infrastructure.Persistence.DtoDbContext";
+        var entities = solution.Domain.Entities.Where(e => !e.IsOwnedEntity).ToList();
+        
+        new TemplateCodeBuilder(context, codeGeneratorState)
+            .WithClassName(className)
+            .WithFileNamePrefix($"Infrastructure.Persistence")
+            .WithObject("RootNameSpace", codeGeneratorState.RootNameSpace)
```
Comment (rochar) [inline src/Nox.Generator/Infrastructure/Persistence/DtoDbContextGenerator.cs]: no need this object use codeGeneratorState.RootNameSpace directly on the template

File: src/Nox.Generator/Infrastructure/Persistence/DtoDbContext.template.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
+// Generated
+#nullable enable
+
+using Microsoft.EntityFrameworkCore;
+using Nox;
+using Nox.Solution;
+using Nox.Types.EntityFramework.Abstractions;
+using {{RootNameSpace}}.Application.Dto;
+
+namespace {{RootNameSpace}}.Infrastructure.Persistence;
```
Comment (rochar) [inline src/Nox.Generator/Infrastructure/Persistence/DtoDbContext.template.cs]:
```suggestion
namespace {{codeGeneratorState.RootNameSpace}}.Infrastructure.Persistence;
```

File: src/Nox.Generator/Infrastructure/Persistence/DtoDbContext.template.cs [hunk 3]
```diff
@@ -0,0 +1,84 @@
+// Generated
+#nullable enable
+
+using Microsoft.EntityFrameworkCore;
+using Nox;
+using Nox.Solution;
+using Nox.Types.EntityFramework.Abstractions;
+using {{RootNameSpace}}.Application.Dto;
```
Comment (rochar) [inline src/Nox.Generator/Infrastructure/Persistence/DtoDbContext.template.cs]:
```suggestion
using {{codeGeneratorState.RootNameSpace}}.Application.Dto;
```

**Review by rochar — APPROVED** [review]
just review my comments

**Review by gjorgji-grgovski-authoritypartners — APPROVED** [review]

File: src/Nox.Generator/Infrastructure/Persistence/DtoDbContext.template.cs (hunk 3 above)
Comment (msadikkoseaupa) [reply src/Nox.Generator/Infrastructure/Persistence/DtoDbContext.template.cs]: Desired changes were applied

File: src/Nox.Generator/Infrastructure/Persistence/DtoDbContext.template.cs (hunk 2 above)
Comment (msadikkoseaupa) [reply src/Nox.Generator/Infrastructure/Persistence/DtoDbContext.template.cs]: Desired changes were applied

File: src/Nox.Generator/Infrastructure/Persistence/DtoDbContextGenerator.cs (hunk 1 above)
Comment (msadikkoseaupa) [reply src/Nox.Generator/Infrastructure/Persistence/DtoDbContextGenerator.cs]: Desired changes were applied

**Pull request merged**